spec/acceptance/keystore_spec.rb

```diff
@@ -57,42 +52,65 @@
         }
       MANIFEST
 
-      apply_manifest(pp_two, catch_failures: true)
-      apply_manifest(pp_two, catch_changes: true)
+      idempotent_apply(default, pp_two)
+    end
+
+    it 'recreates a keystore if password fails' do
+      pp_three = <<-MANIFEST
+
+        java_ks { 'puppetca:keystore':
+          ensure              => latest,
+          certificate         => "#{@temp_dir}ca.pem",
+          target              => '#{target}',
+          password            => 'pepput',
+          password_fail_reset => true,
+          trustcacerts        => true,
+          path                => #{@resource_path},
+      }
+      MANIFEST
+
+      idempotent_apply(default, pp_three)
+    end
+
+    it 'verifies the keystore again' do
+      shell("\"#{@keytool_path}keytool\" -list -v -keystore #{target} -storepass pepput") do |r|
+        expect(r.exit_code).to be_zero
+        expectations.each do |expect|
+          expect(r.stdout).to match(expect)
+        end
+      end
     end
   end
 
-  unless fact('operatingsystemmajrelease') == '18.04'
+  unless os[:family] == 'debian' && os[:release].start_with?('18.04')
     describe 'storetype' do
+      target = "#{@target_dir}storetypekeystore.ks"
+
       it 'creates a keystore' do
         pp = <<-MANIFEST
           java_ks { 'puppetca:keystore':
             ensure       => latest,
             certificate  => "#{@temp_dir}ca.pem",
             target       => '#{target}',
-            password     => 'puppet',
+            password     => 'pepput',
+            password_fail_reset => true,
             trustcacerts => true,
             path         => #{@resource_path},
             storetype    => 'jceks',
           }
         MANIFEST
 
-        apply_manifest(pp, catch_failures: true)
-        apply_manifest(pp, catch_changes: true)
+        idempotent_apply(default, pp)
       end
 
       expectations = [
-        %r{Your keystore contains 2 entries},
+        %r{Your keystore contains 1 entry},
```

Comment:
how come there is only a single entry now ?

Reply:
The password fails test wipes out the second keystore puppetca2 which was previously left behind. I double checked the stdout and it look's correct.